docs/protect_payload.md

```diff
@@ -0,0 +1,134 @@
+# Security Guarantees of the "Strict Payload Policy"
+
+### Threat Model
+
+In this scenario, the threat comes from the firmware, which has the ability to execute arbitrary code, trigger interrupts, and potentially read or modify payload memory. As per our assumptions, side-channel attacks, transient execution vulnerabilities, and CPU bugs are considered out of scope. Finally, we place trust in Miralis and assume it to be free of bugs.
+
+### Guarantees
+
+The strict payload policy ensures two key guarantees:
+
+1. **Confidentiality:** The firmware cannot leak any information from the payload. This ensures that the execution of the payload remains secret to the firmware.
+
+2. **Integrity:** The firmware cannot alter the architectural state of the payload in any way.
+
+### Assumptions
+
+- Miralis is a trusted component and is free of bugs.
+- Side-channel attacks are out of scope.
+- The underlying hardware is trusted and behaves correctly.
+- The strict payload policy is always enforced and "locked."
+- The firmware does not alter the set of CSR registers and Miralis enforces interrupt delegation to S-mode.
+
+### Formal Definitions
+
+A **payload state** is defined as a tuple $S = (\text{memory}, \text{general purpose registers}, \text{csr registers})$, where:
+- `memory` is an ordered set representing memory locations,
+- `general purpose registers` is an ordered set of 32 general-purpose registers,
+- `csr registers` is an ordered set of control and status registers (CSRs).
+
+Two payload states are equivalent if and only if:
+- $\text{memory}_1 \equiv \text{memory}_2$,
+- $\text{gp}_1 \equiv \text{gp}_2 $, and
+- $\text{csr}_1 \equiv \text{csr}_2$.
+### Expected architectural transitions per interrupts / traps - wip
+
+#### Interrupts
+
+- Supervisor software interupt: equivalent state
+- Machine software interup: equivalent state
+- Supervisor timer interrupt: equivalent state
+- Machine timer interrupt: equivalent state
+- Supervisor external interrupt: equivalent state
+- Machine external itnerrupt: equivalent state
+
+#### Traps
+
+- Instruction address misaligned: stop execution?
+- Instruction access fault: stop execution?
+- Illegal instruction: pc += 4
+- Breakpoint: equivalent state
+- Load address misaligned: stop execution?
+- Load access fault: stop execution?
+- Store/AMO address misaligned: stop execution?
+- Ecall from S-mode: pc += 4
+- Instruction page fault: We must assert it happens in S-mode
+- Load page fault: We must assert it happens in S-mode
+- Store/AMO page fault: We must assert it happens in S-mode
+
+
+### State Transitions
+
+The following diagram illustrates the state transitions in our model:
+
+![State Transitions](transitions.png)
```

Comment:
We don't want to store images in the main Miralis repo (in fact the CI checks that there is no big object in any of the commits). For what is in the doc we usually do ascii art for the diagrams.
It's not super important, we can also remove the diagram from the docs, as you prefer :)

Reply:
Btw why this? It doesn't make sense for me. Also images can be compressed and have a small memory footprint

Reply:
The size of a repo only ever grow linearly with time. Images tend to change/being replaced and that creates a lot of footprint in the `.git`. Plus git know how to compress texts efficiently, but not arbitrary images/binary, they just create huge diffs.

If you look at the Ci that tests this branch the two biggest git objects of the whole repo are the two images in this PR:

```
Biggest git objects:
 268958  docs/ace-miralis.png
  85385  docs/transitions.png
  65404  src/virt.rs
  49179  src/arch/metal.rs
  35589  src/decoder.rs
```

Those will basically become dead weight every time we update them (there is no incremental diff for images).